File: frontend-app/disaster_app/lib/models/disaster.dart
class Disaster {
  final int id;
  final String type;
  final double latitude;
  final double longitude;
  final double radius;
  final String message;

  Disaster({
    required this.id,
    required this.type,
    required this.latitude,
    required this.longitude,
    required this.radius,
    required this.message,
  });

  factory Disaster.fromJson(Map<String, dynamic> json) {
    return Disaster(
      id: json['id'],
      type: json['type'],
      latitude: json['latitude'],
      longitude: json['longitude'],
      radius: json['radius'],
      message: json['message'],
    );
  }
}

File: backend/src/main/java/com/disaster/backend/entity/Disaster.java
package com.disaster.backend.entity;

import jakarta.persistence.*;

@Entity
public class Disaster {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String type;
    private double latitude;
    private double longitude;
    private double radius;
    private String message;

    public Disaster() {}

    public Long getId() {
        return id;
    }

    public String getType() {
        return type;
    }

    public void setType(String type) {
        this.type = type;
    }

    public double getLatitude() {
        return latitude;
    }

    public void setLatitude(double latitude) {
        this.latitude = latitude;
    }

    public double getLongitude() {
        return longitude;
    }

    public void setLongitude(double longitude) {
        this.longitude = longitude;
    }

    public double getRadius() {
        return radius;
    }

    public void setRadius(double radius) {
        this.radius = radius;
    }

    public String getMessage() {
        return message;
    }

    public void setMessage(String message) {
        this.message = message;
    }
}
